core/src/main/java/io/aiven/klaw/service/EnvsClustersTenantsControllerService.java

```diff
@@ -886,6 +896,7 @@ public ApiResponse deleteEnvironment(String envId, String envType) throws KlawEx
     }
 
     try {
+      removeEnvironmentMapping(envId, tenantId, envType);
```

Comment:
same here, if its applicable only for schema envs

Reply:
Removing needs to take into account when a kafka env is deleted.
So if a Kafka env is deleted it needs to remove all links between any associated schema env.
Also if the Schema Env is deleted it needs to remove the associated Kafka Env entry.